spec/models/conference_spec.rb

```diff
@@ -778,6 +778,33 @@
     end
   end
 
+  describe '#scheduled_event_distribution' do
+    let!(:conference) { create(:conference) }
+    let!(:selected_schedule) { create(:schedule, program: conference.program) }
+    let!(:confirmed_unscheduled_event) { create(:event, program: conference.program, state: 'confirmed') }
+    let!(:confirmed_scheduled_event) { create(:event, program: conference.program, state: 'confirmed') }
+    let!(:event_schedule) { create(:event_schedule, event: confirmed_scheduled_event, schedule: selected_schedule) }
+
+    before(:each) do 
+      conference.program.update_attributes!(selected_schedule: selected_schedule) 
```

Comment:
if you don't give the `event_schedule` a schedule when creating it, the selected schedule will be automatically created (check [EventSchedule factory](https://github.com/openSUSE/osem/blob/master/spec/factories/event_schedule.rb)) and then you don't need this line, neither the `selected_schedule` creation (line 783).